reconcile/gitlab_labeler.py

```diff
@@ -18,18 +74,35 @@ def guess_labels(project_labels, changed_paths):
     """
     not_allowed_labels = MERGE_LABELS_PRIORITY + HOLD_LABELS
     ignore_tokens = ['cicd', 'saas', 'rds', 'services']
+
+    apps = get_app_list()
+    parent_apps = get_parents_list()
+
     guesses = set()
-    for label in project_labels:
-        if label in not_allowed_labels:
-            continue
-        for path in changed_paths:
-            path_dir = os.path.dirname(path)
-            path_tokens = path_dir.split('/')
-            matches = [t for t in path_tokens if t
-                       and t in label
-                       and t not in ignore_tokens]
-            if matches:
-                guesses.add(label)
+    tenants = []
+    matches = []
+
+    for path in changed_paths:
+        path_dir = os.path.dirname(path)
+        path_tokens = path_dir.split('/')
+
+        tenants += [t for t in path_tokens if t not in ignore_tokens and
+                    t in apps]
+        matches += [t for t in path_tokens if t not in ignore_tokens and
+                    t in project_labels]
+
+    for t in tenants:
+        if t not in not_allowed_labels:
+            guesses.add("tenant-" + t)
+
+    for m in matches:
+        if m not in not_allowed_labels:
+            guesses.add(m)
+
+    onboarding_status = guess_onboarding_status(changed_paths, apps,
+                                                parent_apps)
+    if onboarding_status is not None:
```

Comment:
nit: `guess_onboarding_status` is going return either a string or none... I guess you just need to do
```
if onboarding_status:
```